tools/perf/builtin-c2c.c

```diff
@@ -2547,9 +3124,804 @@ static void perf_c2c__hists_fprintf(FILE *out, struct perf_session *session)
+					nd_cl = rb_next(nd_cl);
+					continue;
+				}
+
+				/* Check if both symbols accessed this cacheline */
+				next_detail = rb_first_cached(&c2c_he_cl->hists->hists.entries);
+				while (next_detail) {
+					struct hist_entry *he_detail = rb_entry(next_detail, struct hist_entry, rb_node);
+					struct c2c_hist_entry *c2c_he_detail = container_of(he_detail, struct c2c_hist_entry, he);
+
+					/* Match by BOTH symbol and code address (iaddr) within the same cacheline */
+					{
+						uint64_t iaddr_detail = he_detail->mem_info ?
+							mem_info__iaddr(he_detail->mem_info)->addr : 0;
+						uint64_t parent_iaddr = he_sym->mem_info ?
+							mem_info__iaddr(he_sym->mem_info)->addr :
+							(he_sym->ms.sym ? he_sym->ms.sym->start : 0);
+
+						if (he_detail->ms.sym == he_sym->ms.sym &&
+						    iaddr_detail == parent_iaddr) {
+							target_found = true;
+							c2c_add_stats(&target_stats, &c2c_he_detail->stats);
+							c2c_add_cstats(&target_cstats, &c2c_he_detail->cstats);
+						} else if (he_detail->ms.sym == rel_sym->sym &&
+						           iaddr_detail == rel_sym->iaddr) {
+							related_found = true;
+							c2c_add_stats(&related_stats, &c2c_he_detail->stats);
+							c2c_add_cstats(&related_cstats, &c2c_he_detail->cstats);
+						}
+					}
+
+					next_detail = rb_next(&he_detail->rb_node);
+				}
+
+				/* If both symbols accessed this cacheline, add the related symbol's stats */
+                if (target_found && related_found) {
+                    c2c_add_stats(&rel_sym->stats, &related_stats);
+                    c2c_add_cstats(&rel_sym->cstats, &related_cstats);
+				}
+
+				nd_cl = rb_next(nd_cl);
+			}
+		}
+
+		nd_sym = rb_next(nd_sym);
+	}
+
+	/* Phase 3: Create child entries for symbols with associations */
+	nd_sym = rb_first_cached(&c2c.symbol_hists.hists.entries);
+	while (nd_sym) {
+		he_sym = rb_entry(nd_sym, struct hist_entry, rb_node);
+		if (he_sym->has_children) {
+			populate_symbol_children(he_sym);
+		}
+		nd_sym = rb_next(nd_sym);
+	}
+}
+
+/* Structure to aggregate symbol stats */
+struct symbol_entry {
+	uint64_t iaddr;
+	struct symbol *sym;
+	struct map *map;
+	struct maps *maps;
+	struct c2c_stats stats;
+	struct compute_stats cstats;
+	uint64_t samples;
+	struct symbol_entry *next;
+};
+
+/* Simple hash table for symbol aggregation */
+#define SYMBOL_HASH_SIZE 1024
+static struct symbol_entry *symbol_hash[SYMBOL_HASH_SIZE];
+
+static unsigned int symbol_hash_func(uint64_t iaddr, struct symbol *sym)
+{
+	return (iaddr ^ (unsigned long)sym) % SYMBOL_HASH_SIZE;
+}
+
+static struct symbol_entry *find_or_create_symbol_entry(uint64_t iaddr, struct symbol *sym, struct map *map, struct maps *maps)
+{
+	unsigned int hash = symbol_hash_func(iaddr, sym);
+	struct symbol_entry *entry;
+
+	/* Search for existing entry */
+	for (entry = symbol_hash[hash]; entry; entry = entry->next) {
+		if (entry->iaddr == iaddr && entry->sym == sym) {
```

Comment:
what's the primary type of `sym`, 
for `entry->sym == sym`, is the pointer cmp, or string cmp?